File: 백준/Silver/10773. 제로/제로.js
const fs = require("fs");
const path = require("path");
const input = fs
  .readFileSync(
    process.platform === "linux"
      ? "/dev/stdin"
      : path.join(__dirname, "/input.txt"),
  )
  .toString()
  .trim()
  .split("\n");

const solution = (input) => {
  const K = Number(input[0]);
  const stack = [];

  for (let i = 1; i <= K; i++) {
    const n = Number(input[i]);
    if (n === 0) {
      stack.pop();
    } else {
      stack.push(n);
    }
  }
  console.log(stack.reduce((acc, cur) => acc + cur, 0));
};

solution(input);

File: 백준/Silver/10773. 제로/제로.cc
#include <iostream>
#include <stack>

using namespace std;

int main()
{
    int K;
    cin >> K;

    stack <int> s;
    for(int i=0;i<K;i++){
        int n;
        cin >> n;
        if(n==0){
            s.pop();
        }else{
            s.push(n);
        }
    }

    int sum=0;
    while(s.size()){
        sum+=s.top();
        s.pop();
    }

    cout << sum;
    return 0;
}